api/src/main/java/com/google/sps/SendNotifications.java

```diff
@@ -1,117 +1,155 @@
-    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
-
-    System.out.println("Google Credentials created...");
-
-    String projectId = "phoebeliang-step";
-    FirebaseOptions options = new FirebaseOptions.Builder().setCredentials(credentials).setProjectId(projectId).build();
-    FirebaseApp.initializeApp(options);
-
-    System.out.println("App Initialized...");
+  public List<EmailCreation> gatherRecipients(HttpServletRequest request, HttpServletResponse response)
+    throws IOException {
 
     Firestore db = FirestoreClient.getFirestore();
-
-    String gameID = "dT6U1wQKR1rGJcEncyMS";
-    List<Email> emails = new ArrayList<>();
+    String gameID = request.getParameter("gameID");
+    EmailType emailType = EmailType.valueOf(request.getParameter("emailType").toUpperCase());
+    List<EmailCreation> emails = new ArrayList<>();
 
     DocumentReference game = db.collection("games").document(gameID);
-
+    CollectionReference players = db.collection("users");
     try {
       DocumentSnapshot gameDocSnap = game.get().get();
-      List<String> playerNames = (List<String>) gameDocSnap.get("players");
 
-      for(String player: playerNames) {
-        ApiFuture<QuerySnapshot> futurePlayers = db.collection("users").whereEqualTo("username", player).get();
-        List<QueryDocumentSnapshot> usersDocSnap = futurePlayers.get().getDocuments();
+      List<String> playerNames = ((List<String>) gameDocSnap.get("players"));
 
-        // Retrieves player information
-        Map<String,Object> playerInfo = usersDocSnap.get(0).getData();
-        String playerEmail = (String) playerInfo.get("email");
-        String playerName = (String) playerInfo.get("username");
+      for (String player : playerNames) {
+        DocumentSnapshot usersDocSnap = players.document(player).get().get();
 
-        System.out.println(playerEmail);
+        // Retrieves player information
+        String playerEmail = usersDocSnap.getString("email");
+        String hostName = players.document(playerNames.get(0)).get().get().getString("username");
 
         // Adding player to Email object
-        emails.add(new Email(new User(playerEmail, playerName)));
-
-        // Adds Start game message to the email object
-        if(emails.isEmpty()) {
+        emails.add(emailType.createEmail(gameID, playerEmail, hostName));
+        
+        // Checks if there are any players
+        if (emails.isEmpty()) {
           throw new IllegalArgumentException();
-        } else {
-          emails.get(emails.size()-1).playerTurn(gameID);
         }
       }
     } catch (Exception e) {
-      System.out.println(e);
+      System.out.println("Exception in gatherRecipients" + e);
     }
 
     return emails;
   }
 
-  @GetMapping("/notifyTurn")
-  public void playerTurnEmail() throws IOException {
-    List<Email> emails = gatherRecipients();
+  public EmailCreation getNextPlayer(HttpServletRequest request, HttpServletResponse response)
```

Comment:
Might consider another name, getNextPlayer() generally are expected to return a Player object.